editor/effects.js

```diff
@@ -210,4 +221,28 @@ export default {
 			]
 		) );
 	},
+	AUTOSAVE( action, store ) {
+		const { dispatch, getState } = store;
+		const state = getState();
+		if ( ! isEditedPostSaveable( state ) || ! isEditedPostDirty( state ) ) {
+			return;
+		}
+
+		if ( isEditedPostPublished( state ) ) {
+			// TODO: Publish autosave
+		} else {
+			dispatch( savePost() );
+		}
+	},
+	QUEUE_AUTOSAVE: debounce( ( action, store ) => {
+		store.dispatch( autosave() );
+	}, 10000 ),
+	UPDATE_BLOCK_ATTRIBUTES: () => queueAutosave(),
+	INSERT_BLOCKS: () => queueAutosave(),
+	MOVE_BLOCKS_DOWN: () => queueAutosave(),
+	MOVE_BLOCKS_UP: () => queueAutosave(),
+	REPLACE_BLOCKS: () => queueAutosave(),
+	REMOVE_BLOCKS: () => queueAutosave(),
+	EDIT_POST: () => queueAutosave(),
+	MARK_DIRTY: () => queueAutosave(),
```

Comment:
>Why do we need all these effects? I suspect dirtiness? Can't we use an internal somewhere instead?

Yeah, all the potentially dirty-ing effects, see also: reducer for `editor.dirty`.

As remarked in #1967, I'd like to look to reimplementing dirtiness as a diff against the original saved post, in which case we could reimplement this as a subscribe on the store (i.e. `if ( currentDirty !== lastDirty ) queueAutosave()`).